rts/Game/UI/IKeys.cpp

```diff
@@ -0,0 +1,132 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+#include "IKeys.h"
+#include "System/StringUtil.h"
+
+
+int IKeys::GetCode(const std::string& name) const
+{
+	const auto pred = [](const NameCodePair& a, const NameCodePair& b) { return (a.first < b.first); };
+	const auto iter = std::lower_bound(nameToCode.begin(), nameToCode.end(), NameCodePair{name, 0}, pred);
```

Comment:
If `nameToCode` is not too big you might also just use `std::fihd`

Reply:
This is how big you can expect it to be: https://github.com/badosu/spring/blob/a4abeaf6e5b9df719f367ea0066603ed26ea06aa/rts/Game/UI/KeyCodes.cpp#L58-L145

You can see the list by typing `/keysyms`, users can add personal aliases to sdl keycodes but most of the time this list is static.

Reply:
I decided to not change this for now